### PR TITLE
Feat/remove ethereumjs

### PR DESCRIPTION
**Pull request by tuna1207**
## Motivation and Context

Remove ethereumjs/tx and /common

Jira Link:

## Description

Replace ethereumjs/tx and /common with ethers to reduce package size

## How has this been tested?

tested with demo app

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project. (run lint)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/formatter.ts [hunk 1]
```diff
@@ -156,7 +142,7 @@ export class TransactionFormatter {
       clonedTxParams.gasPrice = defaultGasPrice as never;
     }
 
-    clonedTxParams.type = this.isEIP1559Compatible ? TRANSACTION_ENVELOPE_TYPES.FEE_MARKET : TRANSACTION_ENVELOPE_TYPES.LEGACY;
+    clonedTxParams.type = Number(this.isEIP1559Compatible ? TRANSACTION_ENVELOPE_TYPES.FEE_MARKET : TRANSACTION_ENVELOPE_TYPES.LEGACY);
```
Comment (chaitanyapotti) [inline packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/formatter.ts:145]: just make this Number.parseInt(x,16) to remove ambiguity

File: packages/providers/ethereum-mpc-provider/src/providers/signingProviders/signingUtils.ts [hunk 2]
```diff
@@ -18,36 +18,12 @@ async function signTx(
...
-    import("@ethereumjs/tx"),
-  ]);
+): Promise<PrefixedHexString> {
+  const { Transaction } = await import("ethers");
   const finalTxParams = await txFormatter.formatTransaction(txParams);
-  const common = await txFormatter.getCommonConfiguration();
-  const unsignedEthTx = TransactionFactory.fromTxData(finalTxParams, {
-    common,
-  });
+  const ethTx = Transaction.from(finalTxParams);
```
Comment (chaitanyapotti) [inline packages/providers/ethereum-mpc-provider/src/providers/signingProviders/signingUtils.ts:24]: might want to remove `from` from finalTxParams

File: packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/formatter.ts [hunk 3]
```diff
@@ -104,10 +90,10 @@ export class TransactionFormatter {
 
         clonedTxParams.maxPriorityFeePerGas = bnLessThan(
           typeof defaultMaxPriorityFeePerGas === "string" ? stripHexPrefix(defaultMaxPriorityFeePerGas) : defaultMaxPriorityFeePerGas,
-          typeof clonedTxParams.gasPrice === "string" ? stripHexPrefix(clonedTxParams.gasPrice) : clonedTxParams.gasPrice
+          typeof clonedTxParams.gasPrice === "string" ? stripHexPrefix(clonedTxParams.gasPrice) : clonedTxParams.gasPrice.toString()
```
Comment (chaitanyapotti) [inline packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/formatter.ts:93]: what's the other type of gasPrice? number, bigint or BN?

Comment (tuna1207) [reply packages/providers/ethereum-provider/src/providers/privateKeyProviders/TransactionFormatter/formatter.ts:93]: @chaitanyapotti it's this one `number | bigint`
```typescript
/**
 *  Any type that can be used where a numeric value is needed.
 */
export type Numeric = number | bigint;

/**
 *  Any type that can be used where a big number is needed.
 */
export type BigNumberish = string | Numeric;
```

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**